### ISSUE TITLE
Postgress is in UTC Timezone which causes timestamp errors in tasks if nautobot runs in a different time zone, timezone should be defined in the environment section for Postgress.

### ISSUE THREAD

**Issue by itheodoridis** [nautobot/nautobot-docker-compose#45]
If the Timezone is not UTC and an appropriate different timezone is defined for nautobot in the nautobot_config.py (or env file) and also by the use of 
```
    volumes:
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro    
```
for each service, the onboarding tasks are mentioned to be running for a wrong amount of time and are timestamped a few hours away from the local time marked in the footer of the nautobot webpage.
The Correct timezone should be defined in the environment area of the Nautobotm docker-compose.yml file for the postgress service:
```
    environment:
      - TZ=Europe/Athens
```
This should be mentioned in the documentation for the repo.

**Comment by jifox** [comment]
@itheodoridis In my opinion the database should stay on UTC.  So one can always be sure that timestamps are always UTC even when restoring from a database from e.g a branch office in another timezone.

Maybe it will be better to define a timezone per user that will be used for displaying times in UI

**Comment by itheodoridis** [comment]
Thank you for the reply.
I would say that opinions are always welcome and possibly debatable with other opinions. 
However what I am saying here is that if a different timezone is set in the docker-compose/nautobot_config.py/local.env combo through the environment variable (which is a supported option) and the DB doesn't follow in its configuration, there are anomalies in the time displayed (like negative time durations for tasks etc).
All my users are in the same timezone so I am definitely proceeding with the change. 
It might be a good idea to note everything in the documentation as it's not describing the whole concept. Perhaps the opinion you expressed should also be included as the recommended approach. 
What I want to avoid is people not being aware of the potential problem.
In my case I want everything in local time unless there is a specific issue with part of the infrastructure which is hardcoded in UTC (as I had in the past issues with InfluxDB & Grafana for example).
Having everything in local time helps with the central logging and other aspects as you can't really transfer the additional calculation (UTC + difference) wherever you transmit data. In a closed system I would agree with you.

**Comment by itheodoridis** [comment]
Well.. I spent a few hours today doing tests at a test environment at work both with real data (synced with the prod system) but also with empty databases. I can't replicate the effect I had at the home lab. I can't reach to a conclusion as to why. Maybe the interaction with actual ntp produced at work is what makes a difference (I don't really believe that, I would like that to be true..).
I have to agree with you on this, you were right. The DB has to remain in UTC. 
The bad thing about this is that it's not enough in a docker based environment to let the DB be configured with UTC (in the DB config where UTC is default). In this environment (probably where ntp comes in) the container also needs to be in UTC for the whole thing to work correctly with Nautobot and have the Nautobot environment display change logs timestamps and job duration correctly.
But that means that I was also right that the container logs may be confusing if put together with logs from other systems in the same timeline. I don't see an elegant solution to this, but this is not something that concerns this project. 
I am closing the issue at this time, thank you for your help!
It's up to you to include any mention on the doc or not, for people using a different timezone with Nautobot on Docker.

**Comment by jvanderaa** [comment]
If you did want to adjust the timestamp within Docker it is pretty quick - https://stackoverflow.com/questions/44394295/how-to-change-timezone-of-postgresql-9-5-on-docker

**Comment by itheodoridis** [comment]
I just tried that. 
```
    environment:
      - TZ=Europe/Athens
      - PGTZ=Europe/Athens
```
It turns out that when you run a job, it gets timestamped as if it's 6 hours earlier (my timezone is currently UTC+3, so it may be a problem of doing substracting a negative value or something like that, which seems like a bug but it's probably by design on the db level when apps and db are operating on the same machine and don't have to handle container environment issues.
So in sort: that doesn't work in total as a solution. I tried several sub-combinations and even managed to crash the test installation. 
So thanks for the suggestion, we are back to no1 option, unless someone wants to create an issue about jobs and how timestamps are calculated.. I think no one does any testing for a different timezone than UTC, I would be thrilled if someone proves me wrong.
I also searched and searched for how to define different timezones per user, as @jifox suggested. I haven't found such an option anywhere.
